assets/src/edit-story/components/form/dateTime/datePicker.js

```diff
@@ -39,12 +44,41 @@ function DatePicker({ currentDate, onChange, onViewChange }) {
     [value, onChange]
   );
 
+  useEffect(() => {
+    // Set tabIndex to -1 for every except for the first button.
+    if (nodeRef.current) {
+      // Allow tabbing to sections inside the calendar.
+      const navButtons = [
+        ...nodeRef.current.querySelectorAll(
+          '.react-calendar__navigation button'
+        ),
+      ];
+      navButtons.shift();
+      for (const btn of navButtons) {
+        btn.tabIndex = '-1';
+      }
+      const buttons = [
+        ...nodeRef.current.querySelectorAll(
+          '.react-calendar__viewContainer button'
+        ),
+      ];
+      buttons.shift();
+      for (const btn of buttons) {
+        btn.tabIndex = '-1';
+      }
+    }
+  }, [nodeRef]);
```

Comment:
This effect unfortunately has to run everytime the component switch view. If you switch to the month view (by clicking the month), all the month buttons now don't have `tabIndex=-1` and navigation breaks again. Clicking a month brings you back to the day view without `tabIndex=-1` and navigation is still broken.

If you move it to a `useCallback` to be called in both the `useEffect` on first render and then also as a callback on every [`onViewChange`](https://github.com/wojtekmaj/react-calendar#user-content-props:~:text=onViewChange)  and [`onActiveStartDayChange`](https://github.com/wojtekmaj/react-calendar#user-content-props:~:text=onActiveStartDateChange) reported from the component, I think it'll work correctly - but it's a bit fragile 😒 .

And maybe the initially selected button (the one without `tabIndex=-1`) shouldn't necessarily be the first one, but rather the currently highlighted one (e.g. current day/month/year) if any.

This is a... meh... but it might work?

Reply:
Made some adjustments based on your recommendation, it seems to work as expected now, PTAL.